contracts/executor/src/lib.rs

```diff
@@ -0,0 +1,263 @@
+                            } else {
+                                // TODO: More validations
+
+                                // Start to execute next step
+                                let signer = PrivateKey(task.worker);
+                                let hash = Step(self.registry)::execute_step(&signer, &task.edges[step_index + 1])?;
+                                task.status = TaskStatus::Executing(step_index + 1, Some(hash));
+                                self.update_task(&task);
+                            }
+                        } else {
+                            // Re-apply nonce
+                            self.apply_worker(&task)?;
+                            // TODO: Retry
+
+                            // TODO: Exceed retry limit, revert
+                            let signer = PrivateKey(task.worker);
+                            let hash = Step(self.registry)::revert_step(&signer, &task.edges[step_index])?;
+                            task.status = TaskStatus::Reverting(step_index, Some(hash));
+                            self.update_task(&task);
+                        }
+                    },
+                    TaskStatus::Reverting(step_index, Some(revert_tx_hash)) => {
+                        let result = TransactionChecker::check_transaction(Step(self.registry)::dest_chain(&task.edges[step_index]), revert_tx_hash)?;
+                        if result {
+                                // TODO: More validations
+
+                                // Start to execute next step
+                                let signer = PrivateKey(task.worker);
+                                let hash = Step(self.registry)::revert_step(&signer, &task.edges[step_index - 1])?;
+                                task.status = TaskStatus::Reverting(step_index - 1, Some(hash));
+                                self.update_task(&task);
+                        } eles {
+                            // Re-apply nonce
+                            self.apply_worker(&task)?;
+                            // TODO: Retry
+
+                            // TODO: Exceed retry limit, task is dead, manual handling???
+                        }
+                    },
+                    TaskStatus::Completed => {
+                        self.remove_task(&task)?;
+
+                        // TODO: update pallet-index
+                    },
+                    _ => {
+                        // Do nothing
+                    }
+                }
+            });
+
+            // 2) Fetch actived task that completed initial confirmation from all supported chains that belong to current worker,
+            // and append them to runing tasks
+            let onchain_actived_tasks = self.registry.supported_chains.iter().map(|chain| {
+                let chain_info = self.registry.chains.get(chain).unwrap();
+                let actived_tasks = ActivedTaskFetcher::new(chain_info, worker).fetch_tasks()?;
```

Comment:
Task creation lifetime cycle:

1. Source: Created
2. Phat: Found
3. Phat: Upload
4. Khala: Created
5. **Source: Claimed** (not before 4 to ensure it will not be dropped)
6. Phat: Executing